src/openklant/components/klantinteracties/api/serializers/actoren.py

```diff
@@ -36,173 +44,161 @@ class Meta:
-                    )
-
-            validated_data["actor"] = Actor.objects.get(uuid=actor_uuid)
-
-        return super().update(instance, validated_data)
+    create_and_update_mapping = {
+        SoortActor.medewerker: MedewerkerSerializer,
+        SoortActor.geautomatiseerde_actor: GeautomatiseerdeActorSerializer,
+        SoortActor.organisatorische_eenheid: OrganisatorischeEenheidSerializer,
+    }
 
     @transaction.atomic
     def create(self, validated_data):
-        actor_uuid = str(validated_data.pop("actor").get("uuid"))
-        actoren = OrganisatorischeEenheid.objects.filter(actor__uuid=actor_uuid)
+        actor_identificatie = validated_data.pop("actor_identificatie")
+        actor = super().create(validated_data)
 
-        if len(actoren) != 0:
-            raise serializers.ValidationError(
-                {"actor.uuid": _("Er bestaat al een actor met eenzelfde uuid.")}
-            )
+        serializer_class = self.create_and_update_mapping[
+            validated_data.get("soort_actor")
+        ]
+        actor_identificatie["actor"] = actor.pk
 
-        validated_data["actor"] = Actor.objects.get(uuid=actor_uuid)
+        serializer = serializer_class(
+            data=actor_identificatie, context=self.context, many=False
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return actor
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        if "actor_identificatie" in validated_data:
+            actor_identificatie = validated_data.pop("actor_identificatie")
+            serializer_class = self.create_and_update_mapping[
+                validated_data.get("soort_actor")
+            ]
+            actor_identificatie["actor"] = instance.pk
+
+            try:
+                polymoprhic_instance = serializer_class.Meta.model.objects.get(
+                    actor=instance
+                )
+                serializer = serializer_class(
+                    data=actor_identificatie,
+                    context=self.context,
+                    instance=polymoprhic_instance,
+                    many=False,
+                )
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+            except serializer_class.Meta.model.DoesNotExist:
+                self.context["request"].method = "POST"
+                serializer = serializer_class(
+                    data=actor_identificatie,
+                    context=self.context,
+                    many=False,
+                )
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+
+                # Delete old record
+                serializer_class = self.create_and_update_mapping[instance.soort_actor]
+                polymoprhic_instance = serializer_class.Meta.model.objects.get(
+                    actor=instance
+                )
+                polymoprhic_instance.delete()
```

Comment:
Dit ziet er complexer uit dan dit: https://github.com/open-zaak/open-zaak/blob/main/src/openzaak/components/zaken/api/serializers/zaakobjecten.py#L254-L278

Waarom is dat?